src/components/core/MultiStepButton.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { FC, useState, useCallback } from 'react';
+import styled from 'styled-components';
+
+import { BubbleButton, UnstyledButton } from './Button';
+
+enum MultiStepState {
+  DEFAULT,
+  APPROVE,
+}
+
+const { APPROVE, DEFAULT } = MultiStepState;
+
+interface Props {
+  className?: string;
+  needsUnlock?: boolean;
+  valid: boolean;
+}
+
+const Container = styled.div`
```

Comment:
As a general pattern, you might want to define styled components from top to bottom, innermost to outermost. The reason for this is that it allows you to target child components from parents, which is especially useful for overriding styles based on props or via a classname.